java-checks/src/main/java/org/sonar/java/checks/spring/StatusCodesOnResponseCheck.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.spring;
+
+import java.util.List;
+import java.util.stream.Stream;
+import org.sonar.check.Rule;
+import org.sonar.java.model.ExpressionUtils;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.ClassTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.MemberSelectExpressionTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S6863")
+public class StatusCodesOnResponseCheck extends IssuableSubscriptionVisitor {
+
+  public static final String RESPONSE_ENTITY = "org.springframework.http.ResponseEntity";
+  public static final String OK_ISSUE_MESSAGE = "Use the \"ResponseEntity.ok()\" method or set the status to \"HttpStatus.OK\".";
+  public static final String ERROR_ISSUE_MESSAGE = "Use the \"ResponseEntity.badRequest()\" or \"ResponseEntity.notFound()\" method" +
+    "or set the status to \"HttpStatus.INTERNAL_SERVER_ERROR\" or \"HttpStatus.NOT_FOUND\".";
+
+  private final List<String> okCodes = List.of("CONTINUE",
```

Comment:
We should add a link to the documentation we extracted these codes. This will help us in the future when dealing with FPs/FNs to ensure that we are using a list that is up-to-date (or needs to be updated).

Reply:
Complementary question: Is the list sorted in any way? Alphabetically? According to the error code?